src/main/java/Duke.java

```diff
@@ -1,10 +1,193 @@
+            tasks[numOfDone - 1].markAsDone();
+            System.out.println("___________________________________________________________" + System.lineSeparator()
+                    + "Great work! I've marked this task as done:" + System.lineSeparator()
+                    + "[" + tasks[numOfDone - 1].getType() + "]" + "[" + tasks[numOfDone - 1].getStatusIcon() + "] "
+                    + tasks[numOfDone - 1].description + System.lineSeparator()
+                    + "___________________________________________________________");
+        }
+        else {
+            System.out.println("That is not a valid task number! Please try again.");
+        }
+    }
+
+    public static void addTask(Task t) {
+        tasks[tasksCounter] = t;
+        tasksCounter++;
+    }
+
+    public static void addTodo(String description) {
+        addTask(new Todo(description));
+        printAddedTodo(description);
+    }
+
+    public static void addDeadline(String description, String by) {
+        addTask(new Deadline(description, by));
+        printAddedDeadline(description,by);
+    }
+
+    public static void addEvent(String description, String at) {
+        addTask(new Event(description, at));
+        printAddedEvent(description, at);
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // initialize input
+        Scanner in = new Scanner(System.in);
+        String line;
+
+        // start the taskbot
+        printWelcome();
+        line = in.nextLine();
+
+        // while input is not "bye", keep taking inputs.
+        while (!line.equals("bye")) {
+            if (line.equals("list")) { // print List when "list" command
+                printList();
+            }
+            else if (line.equals ("@help")) { // print help commands when "@help" command
+                printHelp();
+            }
+            else if (line.contains("done")) { // mark task as Done when "done" command
+                int taskNum = filterTaskNum(line);
+                markTaskDone(taskNum);
+            }
+            else if (line.contains("todo")) { // add a todo when "todo" command
+                String todoDescription = line.substring(5);
+                addTodo(todoDescription); // adds todo and prints
+            }
+            else if (line.contains("deadline")) { // add a deadline when "deadline" command
+                int byPos = line.indexOf("/by");
+                if (byPos == -1) { // throw error if missing /by parameter
+                    System.out.println("ERROR! Did you forget /by ?");
+                    break;
+                }
+                int lastPos = line.length();
```

Comment:
maybe renaming lastPos to positionOfLastChar might be clearer, because you are using .length() to find it's index. But lastPos is clear too, just had to double confirm what it meant by looking at .length()